src/pages/Schedule.js

```diff
@@ -1,49 +1,170 @@
+          &&
+          pickedDate.toString() === el.date.date.toString()
+        )
+      } else {
+        return pickedDate.toString() === el.date.date.toString()
+      }
+    });
+
+    this.setState({
+      filteredMatches: filteredMatches,
+      dateOptionValue: pickedDate
+    })
+  }
+
+  handleOnTeamChange = (e) => {
+    const pickedTeam = e.target.value;
+
+    // Display list of scheduled matches for particular team or in case you chose date also, display scheduled matches based on selected date and team
+    const filteredMatches = this.state.scheduledMatches.filter(el => {
+      if (this.state.dateOptionValue.toString() !== "none") {
+
+        if (pickedTeam.toString() === "none") {
+          return this.state.dateOptionValue.toString() === el.date.date.toString()
+        } else {
+          return (
+            this.state.dateOptionValue.toString() === el.date.date.toString()
+            &&
+            (pickedTeam.toString() === el.home_team.id.toString() || pickedTeam.toString() === el.away_team.id.toString())
+          )
+        }
+      } else {
+        return pickedTeam.value.toString() === el.home_team.id.toString() || pickedTeam.toString() === el.away_team.id.toString()
+      }
+    });
+    this.setState({
+      filteredMatches: filteredMatches,
+      teamOptionValue: pickedTeam
+    })
+  }
+  /************************************************************************************************************************************ */
+  // HTML
+  render() {
     return (
       <div className="container">
         <h1 className="display-4 border-bottom mb-4 mt-5">Rozpis</h1>
-        <div className="card">
-          <div className="card-body bg-light">
-            <h4 className="display-6 border-bottom mb-4 ">Deň</h4>
-
-            <select className="custom-select">
-              {
-                this.state.dates.map((dateObj, index) => {
-                  const kslDate = Utils.getDateOption(dateObj.date);
-                  return (
-                    <option key={index} value={index}> {kslDate} </option>
-                  )
-                })
-              }
-            </select>
-          </div>
-        </div>
-        <div className="card mt-4">
-          <div className="card-body bg-info">
-            There will be list of current day matches
+        <div className="container ">
+          <div className="row">
+            <div className="col-lg-6 col-sm-6 col-xs-12">
+              <div className="card">
+                <div className="card-body bg-light">
+                  <h4 className="display-6 border-bottom mb-4 ">Deň</h4>
+                  <select className="custom-select" onChange={this.handleOnDateChange} value={this.state.dateOptionValue}>
+                    {
+                      this.state.dates.map((dateObj, index) => {
```

Comment:
Im thinking if we really need to run this map function each time render function is called. Do we? Because list of option elements is constant, it doesnt change. I think you could run this map function in advance once you got the data and just display the result here